### DISCUSSION TITLE
I want to suppress window maximization

### DISCUSSION THREAD

**Discussion by sdr0x07b6** [tauri-apps/tauri#10305] [Q&A] [answered]
I am trying to create an application for macOS with Tauri v1.
I have `tauri.windows.decorations` set to `false` and have given the entire window container `data-tauri-drag-region`.
When I double click on this window it maximizes, can I suppress the maximization?

I want to be able to resize the window. I want to suppress maximization only.
The approach of automatically minimizing when maximized is not acceptable. Maximization and minimization involve fancy animations by the OS. We want to prevent maximization itself.

Thanks.

**Comment by FabianLars** [comment]
In case this is still an open question, did you ever try the `maximizable: false` config?

**Reply by sdr0x07b6** [reply]
Thank you for your comment.
I tried setting `tauri.windows.maximizable` in `tauri.conf.json` to `false`, but nothing changed.
I wonder if it's a bug in Tauri that allows me to maximize even though `maximizable` is `false`...

**Reply by FabianLars** [reply]
Can you try disallowing the `setMaximized` api in the window allowlist (also in tauri.conf.json)? That _should_ be what the drag region uses so properly disabling that api should hopefully work. If it doesn't i'll have access to my macos system tomorrow and can check it out myself then.

**Reply by sdr0x07b6** [reply]
https://github.com/tauri-apps/tauri/issues/3215
Are you talking about this property?
I tried setting both `maximized` and `fullscreen` to `false` but nothing changed.
```json
"windows": [
  {
    "decorations": false,
    "transparent": true,
    "width": 400,
    "height": 400,
    "resizable": true,
    "maximizable": false,
    "maximized": false,
    "fullscreen": false,
    "titleBarStyle": "Transparent"
  }
]
```

I'm not sure if this is relevant, but my UI is a bit unusual.
My windows don't have standard title bars. Instead, I use `data-tauri-drag-region` to make the entire window draggable. Because the entire window becomes the title bar, double-clicking the window maximizes it. I want to prevent this.

**Reply by FabianLars** [reply]
That's what i meant in my first message. In my last message i meant this: https://github.com/tauri-apps/tauri/blob/1.x/examples/multiwindow/tauri.conf.json#L29-L33
```json
    "allowlist": {
      "all": false,
      "window": {
        "maximize": false
      }
    },
```
(It was `maximize` not `setMaximized`, sorry)

**Reply by sdr0x07b6** [reply]
I found a property called `tauri.allowlist.window.maximize`.
I tried setting it to `false`, but that didn't change the situation. It still maximizes when I double-click.

**Reply by FabianLars** [reply]
Did you set "all": true in `allowlist.window` or `allowlist`? That will overwrite the maximize: false.

If that also doesn't help, could you try to create a minimal reproduction? Or share your project if it's public. I just tried it on my macbook and it seems to work fine. My exact steps to test it were:

1) `pnpm create tauri-app` Using the vanilla-ts framework (really just spammed enter lol)
2) Added `"maximizable": false` to `tauri.windows`
3) Added `"maximize": false` to `tauri.allowlist.window`
4) Added `data-tauri-drag-region` to the existing `h1` element in index.html

**Reply by sdr0x07b6** [reply, marked as answer]
Solved!

When I set both `tauri.allowlist.all` and `tauri.allowlist.window.all` to `false`, it stopped maximizing.

Next, I changed back the parts I had changed. In the end, `tauri.windows.maximizable` and `tauri.allowlist.window.maximizable` did not maximize even though the properties themselves did not exist.

In other words, I just set both `tauri.allowlist.all` and `tauri.allowlist.window.all` to `false`.

Even if `tauri.allowlist.all: true`, if `tauri.allowlist.window.all: false`, it seems that `false` would take precedence only for Window, but in fact both needed to be `false`.

Thank you for your support!
